src/lm/const-arpa-lm.cc

```diff
@@ -278,7 +278,9 @@ void ConstArpaLmBuilder::ConsumeNGram(const NGram &ngram) {
                            cur_order == ngram_order_ - 1,
                            ngram.logprob, ngram.backoff);
 
-    KALDI_ASSERT(seq_to_state_.find(ngram.words) == seq_to_state_.end());
+    if (seq_to_state_.find(ngram.words) != seq_to_state_.end()) {
+      KALDI_ERR << "ngram " << ngram.words << " appears twice in the arpa file";
```

Comment:
This code makes me nervous because you are relying on the existence of a way to print using << a std::vector.
Nothing like that exists in the standard library to my knowledge, and I don't remember defining such a thing.  It may come from OpenFst but who knows whether it will stay forever.  I'd rather you do something manually.